app/qml/misc/PositionProviderPage.qml

```diff
@@ -28,84 +30,266 @@ Page {
+      MouseArea {
+        anchors.fill: parent
+        onClicked: {
+            root.positionKit.positionProvider = root.positionKit.constructProvider( model.ProviderType, model.ProviderId )
+        }
+      }
+
+      Row {
+        id: row
 
-      Column {
         anchors.fill: parent
-        anchors.leftMargin: 5
-        anchors.topMargin: 5
+        anchors.leftMargin: InputStyle.panelMargin
+        anchors.rightMargin: InputStyle.panelMargin
+        anchors.bottomMargin: 5 * __dp
+        anchors.topMargin: 5 * __dp
+
+        RadioButton {
+          id: isActiveButton
+
+          width: parent.height
+          height: parent.height
+
+          checkable: false
+          checked: __appSettings.activePositionProviderId === model.ProviderId
 
-        Text {
-          id: deviceName
+          indicator: Rectangle {
+            implicitWidth: isActiveButton.height / 2.3
+            implicitHeight: isActiveButton.height / 2.3
 
-          width: parent.width
-          height: parent.height * 0.5
+            anchors.centerIn: isActiveButton
 
-          text: model.DeviceName
+            radius: InputStyle.circleRadius
+            border.color: InputStyle.darkGreen
+
+            Rectangle {
+              width: parent.width / 1.5
+              height: parent.height / 1.5
+
+
+              x: parent.width / 2 - width / 2
+              y: parent.height / 2 - height / 2
+
+              radius: InputStyle.circleRadius
+              color: InputStyle.darkGreen
+              visible: __appSettings.activePositionProviderId === model.ProviderId
+            }
+          }
+
+          // We need to duplicate mouse area here in order to handle clicks from RadioButton
+          MouseArea {
+            anchors.fill: parent
+            onClicked: {
+                root.positionKit.positionProvider = root.positionKit.constructProvider( model.ProviderType, model.ProviderId )
+            }
+          }
         }
 
-        Text {
-          id: deviceAddress
-          width: parent.width
-          height: parent.height * 0.5
+        Column {
+          width: row.width - isActiveButton.width - removeIconContainer.width
+          height: row.height
+
+          Text {
+            id: deviceName
+
+            width: parent.width
+            height: parent.height * 0.5
+
+            text: model.ProviderName ? model.ProviderName : qsTr( "Unknown device" )
+
+            elide: Text.ElideRight
+            color: InputStyle.fontColor
+            font.pixelSize: InputStyle.fontPixelSizeNormal
+          }
 
-          text: model.DeviceAddress + ", signal strength: " + model.SignalStrength
+          Text {
+            id: deviceSecondaryText
+
+            width: parent.width
+            height: parent.height * 0.5
+
+            // add BT address to text if it is external receiver
+            text: model.ProviderDescription + ( model.ProviderType === "external" ? " (" + model.ProviderId + ")": "" )
```

Comment:
I see this multiple times in the qml, maybe have another prop in model like "title" ? with this sting?